specs-go/config.go

```diff
@@ -11,7 +11,7 @@ type Spec struct {
 	// Process configures the container process.
 	Process *Process `json:"process,omitempty"`
 	// Root configures the container's root filesystem.
-	Root Root `json:"root"`
+	Root *Root `json:"root,omitempty"`
```

Comment:
> Does it make sense to have a pointer-required-if-field-must-not-be-set-on-a-platform policy instead?

We already [have some policy wording for pointers][1], but they focus on input requirements.  I think we also want to consider output requirements with Go's default JSON serializer, and would recommend:

> Use `omitempty` on properties which  MAY be unset for a platform that uses the parent type.  Of `omitempty` properties, use pointers if the Go zero value is legal (so we can serialize `"uid": 0` and such) or if the property is a Go type (to work around golang/go#11939).

as a new hole we'd want to poke in the [current no-pointer preference][1].  That would mean that we'd want `omitempty` here (because `root` MUST NOT be set for Hyper-V containers, which passes “MAY be unset for a platform that uses the parent type”) and we'd want a pointer here (because property is a Go type).  However, it would also mean we'd want `omitempty` for `User.UID` and `User.GID` (because they MAY be unset on Windows, and Windows uses `User` too for `User.Username`) and we'd want pointers for them (because `0` is a valid ID), and those changes were rejected in #759.  So instead of proposing a pointer policy, I'm just trying to flag changes that may be impacted by the maintainers apparent policy and waiting for them to us how they want that case handled.

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0-rc5/style.md#optional-settings-should-not-have-pointer-go-types